kDriveCore/Utils/DeeplinkParser.swift

```diff
@@ -0,0 +1,68 @@
+/*
+ Infomaniak kDrive - iOS App
+ Copyright (C) 2023 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import InfomaniakDI
+import SwiftUI
+
+public struct DeeplinkParser {
+    @LazyInjectService var accountManager: AccountManageable
+    @LazyInjectService var navigationManager: NavigationManageable
+
+    private let window: UIWindow?
+
+    public init(window: UIWindow?) {
```

Comment:
Having a `DeeplinkParser` that requires a reference to UIKit elements seems strange.
Since we do not have an abstracted navigation, I guess we cannot change this without refactoring the navigationManager.

Reply:
I think we can solve this with a new protocol that can provide the `rootViewController` instead of passing the window as an argument. That way the code would be more abstract, what to you think @Ambrdctr  ?

Reply:
Or we should move the view related code to the `navigationManager` and pass it an enum to remove the view code from this type.

Reply:
Abstracted navigation and unit tested in last commit.